File: ArrayApi.java
public class ArrayApi
{
   public static long sort(int len)
   {
      int[] arr = new int[len];
      for (int i = 0; i < arr.length; i++)
      {
         arr[i] = (int)(Math.random() * len);
         if (i % 2 == 0)
         {
            arr[i] = -arr[i];
         }
      }

      long start = System.currentTimeMillis();
      java.util.Arrays.sort(arr);
      long diff = System.currentTimeMillis() - start;

      for (int i = 1 ; i < arr.length ; i++)
      {
         if (arr[i-1] > arr[i])
         {
            throw new AssertionError("not sorted!!");
         }
      }

      return diff;
   }

   public static void main(String[] args)
   {
      int len = 10;

      if (args.length != 1)
      {
         System.err.println("usage: java ArrayApi ARRAYLENGTH");
         System.exit(1);
      }

      len = Integer.parseInt(args[0]);

      long time = Long.MAX_VALUE;
      
      for (int i = 0 ; i < 5 ; i++)
      {
         time = Math.min(time, sort(len));
      }

      System.out.println("java " + ArrayApi.class.getName() + ": " + time + " ms (" + len + ")");
   }
}

File: ArrayApi.cs
using System;
using System.Diagnostics;

public class Quicksort
{
   public static double Sort(int len)
   {
      var arr = new int[len];
      var random = new Random();
      for (int i = 0; i < arr.Length; i++)
      {
         arr[i] = (int)(random.Next(len));
         if (i % 2 == 0)
         {
            arr[i] = -arr[i];
         }
      }

      var start = DateTime.Now;
      Array.Sort(arr);
      var diff = DateTime.Now - start;

      for (var i = 1 ; i < arr.Length ; i++)
      {
		  if (arr[i-1] > arr[i])
		  {
	         Console.WriteLine("not sorted!!");
	         Environment.Exit(1);
		  }
      }

      return diff.TotalMilliseconds;
   }

   public static void Main(string[] args)
   {
      if (args.Length != 1)
      {
         Console.WriteLine("usage: Quicksort ARRAYLENGTH");
         Environment.Exit(1);
      }

      var len = int.Parse(args[0]);

      var time = double.MaxValue;
      
      for (int i = 0 ; i < 5 ; i++)
      {
         time = Math.Min(time, Sort(len));
      }

      Console.WriteLine("C# Array.Sort() api: " + time + " ms (" + len + ")");
   }
}